### PR TITLE
Add redream cask

### PR DESCRIPTION
**Pull request by larsenv**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).
- [x] `brew audit --cask {{cask_file}}` is error-free.
- [x] `brew style --fix {{cask_file}}` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew install --cask {{cask_file}}` worked successfully.
- [x] `brew uninstall --cask {{cask_file}}` worked successfully.

### CODE REVIEW

**Review by vitorgalvao — CHANGES_REQUESTED** [review]
Welcome.

Please add a [`zap`](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md). See [`homebrew-createzap`](https://github.com/nrlquaker/homebrew-createzap) for an external project which can help.

File: Casks/redream.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask "redream" do
+  version "1.5.0"
+  sha256 "e5253527a3705b4c99234ddf8ece59d19dfadb416822ad714f7009b82afc097b"
+
+  url "https://redream.io/download/redream.x86_64-mac-v#{version}.tar.gz"
+  name "redream"
```
Comment (vitorgalvao) [inline Casks/redream.rb:6]:
```suggestion
  name "Redream"
```

**Comment by larsenv** [comment]
@vitorgalvao is zap just to uninstall program settings?

**Comment by vitorgalvao** [comment]
> is zap just to uninstall program settings?

From the docs:

> `zap` stanzas may remove:
>
> * Preference files and caches stored within the user’s `~/Library` directory.
> * Shared resources such as application updaters. Since shared resources may be removed, other applications may be affected by `brew uninstall --zap`. Understanding that is the responsibility of the end user.
>
> `zap` stanzas should not remove:
>
> * Files created by the user directly.

If the user created it as an output of using the program, keep it. Otherwise, delete it.

File: Casks/redream.rb (hunk 1 above)
Comment (larsenv) [reply Casks/redream.rb]: No. The emulator name is officially capitalized as "redream", all lowercase.

**Comment by larsenv** [comment]
OK, I added a zap, sorry for the delay.

File: Casks/redream.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/redream.rb:6]: > The emulator name is officially capitalized as "redream", all lowercase

Officially, it’s bloody inconsistent.

The logo is all uppercase; the app and support files are all lowercase; on the download page only the first letter is uppercase.

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**